protobuf/v1/replica.proto

```diff
@@ -83,18 +83,16 @@ message ListReplicasResponse {
   repeated Replica replicas = 1;  // list of the replicas
 }
 
-/// ReplicaType enum is used to filter replica in listreplica response.
-enum ReplicaType {
-  AllReplicas  = 0; // Default value, to return all the replicas including snapshots and clones
-  AllReplicasExceptSnapshots = 1; // Return replicas and snapshot clones, without snapshots
-  OnlySnapshotClones = 2; // Returns only snapshot clones
-  OnlyReplicas = 3; // Returns only replicas
-}
-
 message ListReplicaOptions {
   google.protobuf.StringValue name = 1; // list the replica with the name if provided
   google.protobuf.StringValue poolname = 2; // list the replicas on the provided pool, by name
   google.protobuf.StringValue uuid = 3; // list the replica with the uuid if provided
   google.protobuf.StringValue pooluuid = 4; // list the replicas on the provided pool, by uuid
-  ReplicaType replicatype = 5; // list replica based on ReplicaType value
+  Query query = 5; // list replica based on Query value
+
+  /// Query used for getting specific types of replicas.
+  message Query {
+    optional bool snapshot = 1; // query for selecting or snapshot types.
+    optional bool clone = 2; // query for selecting or clone types.
```

Comment:
what is a normal replica here, !clone & !clone?

Reply:
Yes. 
```
ReplicaType::OnlyReplicas =>  !r.is_snapshot && !r.is_clone
```

Reply:
What if in the future we have a new replica type, what will normal replica be?
Maybe we should have an explicit bool for normal replicas?

Reply:
If we have a new type then we would just add that here and normal replica would be !new_type.

Reply:
That's my point, a normal replica would change.
now it's
!r.is_snapshot && !r.is_clone
next it'd be
!r.is_snapshot && !r.is_clone && !r.is_xxx
So we'd have to change every location to include is_xxx: Some(false)
In that case I prefer the current approach tbh

Reply:
I still feel this change would be better as we can have one to one mapping with the specific field. So we can specifically get the ones we want. Other wise we would need a new query type which would be a combination of x y z.

Reply:
It won't be better because it'd be a breaking change for old control plane -> new dataplane
Now: Query: { snap: false, clone: false}
yields replicas
Future with new dataplane with is_xxx and old control-plane:  Query: { snap: false, clone: false}
yields replicas and xxx

Reply:
Why would we have old control-plane? That's not a valid upgrade right?

Reply:
Not as the rule of thumb but we have to consider it may happen, example for CIVO they many times updated components out of order to get specific fixes.